packages/storage/src/AwsClients/S3/headObject.ts

```diff
@@ -64,7 +62,7 @@ const headObjectDeserializer = async (
 ): Promise<HeadObjectOutput> => {
 	if (response.statusCode >= 300) {
 		const error = await parseXmlError(response);
-		throw error;
+		throw StorageError.formServiceError(error, response.statusCode);
```

Comment:
Nit: from previous review:
```suggestion
		throw StorageError.fromServiceError(error, response.statusCode);
```